plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/auth/DefaultAuthCredentialsService.kt

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.amazonq.lsp.auth
+
+import com.intellij.openapi.project.Project
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.AmazonQLspService
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.encryption.JwtEncryptionManager
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.credentials.BearerCredentials
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.credentials.UpdateCredentialsPayload
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.credentials.UpdateCredentialsPayloadData
+import java.util.concurrent.CompletableFuture
+
+class DefaultAuthCredentialsService(
+    private val project: Project,
+    private val encryptionManager: JwtEncryptionManager,
+) : AuthCredentialsService {
+
+    override fun updateTokenCredentials(accessToken: String, encrypted: Boolean): CompletableFuture<ResponseMessage> {
+        val token = if (encrypted) {
+            encryptionManager.decrypt(accessToken)
+        } else {
+            accessToken
+        }
+
+        val payload = createUpdateCredentialsPayload(token)
+
+        return CompletableFuture<ResponseMessage>().also { completableFuture ->
+            AmazonQLspService.executeIfRunning(project) { server ->
+                server.updateTokenCredentials(payload)
+                    .whenComplete { response, throwable ->
+                        if (throwable != null) {
+                            completableFuture.completeExceptionally(throwable)
+                        } else {
+                            completableFuture.complete(response)
+                        }
+                    }
+            } ?: completableFuture.completeExceptionally(IllegalStateException("LSP Server not running"))
+        }
+    }
```

Comment:
i think we can reduce the boilerplate to
```suggestion
        return AmazonQLspService.executeIfRunning(project) { server ->
                server.updateTokenCredentials(payload)
            } ?: CompletableFuture.failedFuture(IllegalStateException("LSP Server not running"))
```

then redefine the execute methods with type generics
```kt
    suspend fun<T> execute(runnable: suspend (AmazonQLanguageServer) -> T): T {
        val lsp = withTimeout(10.seconds) {
            val holder = mutex.withLock { instance }.await()
            holder.initializer.join()

            holder.languageServer
        }

        return runnable(lsp)
    }

    fun<T> executeSync(runnable: suspend (AmazonQLanguageServer) -> T): T =
        runBlocking(cs.coroutineContext) {
            execute(runnable)
        }
```